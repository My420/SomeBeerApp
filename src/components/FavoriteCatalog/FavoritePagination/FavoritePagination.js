import React from 'react';
import PropTypes from 'prop-types';
import styles from './FavoritePagination.module.scss';
import ListGroup from '../../ListGroup/ListGroup';
import ListGroupItem from '../../ListGroupItem/ListGroupItem';
import Button from '../../Button/Button';

const FavoritePagination = ({
  isPrevDisabled,
  isNextDisabled,
  onUserClick
}) => {
  return (
    <ListGroup className={styles.list}>
      <div className={styles.wrapper}>
        <ListGroupItem className={styles.item}>
          <Button
            className={styles.prev}
            onClick={onUserClick}
            data-action="prev"
            disabled={isPrevDisabled}
          >
            prev
          </Button>
        </ListGroupItem>
        <ListGroupItem className={styles.item}>
          <Button
            className={styles.next}
            onClick={onUserClick}
            data-action="next"
            disabled={isNextDisabled}
          >
            next
          </Button>
        </ListGroupItem>
      </div>
    </ListGroup>
  );
};

FavoritePagination.propTypes = {
  isPrevDisabled: PropTypes.bool.isRequired,
  isNextDisabled: PropTypes.bool.isRequired,
  onUserClick: PropTypes.func.isRequired
};

export default FavoritePagination;
