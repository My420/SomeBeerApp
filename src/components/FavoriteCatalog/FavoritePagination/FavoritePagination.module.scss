@import '../../../sass/variables.scss';

.list {
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  height: 70px;

  padding-bottom: 20px;

  list-style: none;
}

.wrapper {
  display: flex;
}

.prev,
.next {
  @include link-button;
}

.next {
  margin-left: 40px;
}

@media (max-width: $Tablet7-max-width) {
  .list {
    height: 30px;

    padding-bottom: 30px;
  }
}
