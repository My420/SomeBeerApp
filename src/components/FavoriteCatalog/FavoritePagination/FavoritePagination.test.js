import React from 'react';
import { shallow } from 'enzyme';
import FavoritePagination from './FavoritePagination';

describe('test <FavoritePagination /> component', () => {
  const mockFn = jest.fn();
  const props = {
    isPrevDisabled: true,
    isNextDisabled: false,
    onUserClick: mockFn
  };

  const wrapper = shallow(<FavoritePagination {...props} />);

  beforeEach(() => {
    mockFn.mockClear();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render prev Button component', () => {
    expect(wrapper.find('Button.prev')).toHaveLength(1);
    expect(wrapper.find('Button.prev').prop('children')).toBe('prev');
  });

  test('should render next Button component', () => {
    expect(wrapper.find('Button.next')).toHaveLength(1);
    expect(wrapper.find('Button.next').prop('children')).toBe('next');
  });

  test('prev Button should have correct disabled attr', () => {
    expect(wrapper.find('Button.prev').prop('disabled')).toBe(
      props.isPrevDisabled
    );
  });

  test('next Button should have correct disabled attr', () => {
    expect(wrapper.find('Button.next').prop('disabled')).toBe(
      props.isNextDisabled
    );
  });

  test('prev Button should have correct data-action attr', () => {
    expect(wrapper.find('Button.prev').prop('data-action')).toBe('prev');
  });

  test('next Button should have correct data-action attr', () => {
    expect(wrapper.find('Button.next').prop('data-action')).toBe('next');
  });

  test('click on next Button should call onUserClick prop', () => {
    wrapper.find('Button.next').simulate('click');
    expect(mockFn.mock.calls.length).toBe(1);
  });

  test('click on prev Button should call onUserClick prop', () => {
    wrapper.find('Button.prev').simulate('click');
    expect(mockFn.mock.calls.length).toBe(1);
  });
});
