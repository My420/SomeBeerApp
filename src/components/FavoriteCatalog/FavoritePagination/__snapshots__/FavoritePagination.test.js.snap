// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <FavoritePagination /> component should match snapshot 1`] = `
<ListGroup
  className="list"
  tag="ul"
>
  <div
    className="wrapper"
  >
    <ListGroupItem
      className="item"
    >
      <Button
        activeStyle=""
        className="prev"
        data-action="prev"
        disabled={true}
        onClick={[MockFunction]}
      >
        prev
      </Button>
    </ListGroupItem>
    <ListGroupItem
      className="item"
    >
      <Button
        activeStyle=""
        className="next"
        data-action="next"
        disabled={false}
        onClick={[MockFunction]}
      >
        next
      </Button>
    </ListGroupItem>
  </div>
</ListGroup>
`;
